tests/unit/lms/models/group_info_test.py

```diff
@@ -43,6 +44,59 @@ def test_application_instance_cant_be_None(self, db_session, group_info):
         ):
             db_session.flush()
 
+    def test_instructors_defaults_to_an_empty_list(self):
+        assert GroupInfo().instructors == []
+
+    def test_upsert_instructor_when_no_existing_instructors(self):
+        group_info = GroupInfo()
+        instructor = factories.HUser()._asdict()
+
+        group_info.upsert_instructor(instructor)
+
+        assert group_info.instructors == [instructor]
+
+    def test_upsert_instructor_when_existing_instructors(self):
+        group_info = GroupInfo()
+        existing_instructors = group_info.instructors = [
+            factories.HUser()._asdict(),
+            factories.HUser()._asdict(),
+            factories.HUser()._asdict(),
+        ]
+        new_instructor = factories.HUser(username="new_instructor")._asdict()
+
+        group_info.upsert_instructor(new_instructor)
+
+        assert group_info.instructors == existing_instructors + [new_instructor]
+
+    def test_upsert_instructor_when_existing_matching_instructor(self):
+        group_info = GroupInfo()
+        updated_instructor = factories.HUser()._asdict()
+        existing_instructors = group_info.instructors = [
+            factories.HUser()._asdict(),
+            factories.HUser(username=updated_instructor["username"])._asdict(),
+            factories.HUser()._asdict(),
+        ]
```

Comment:
I wonder if we could flip this one around and have all three of these test functions have accept "group_with_instructors" which has three instructors set, then make our `updated_instructor` read it's username from the instructor in position 1?

Reply:
Done